gradoop-temporal/src/main/java/org/gradoop/temporal/io/impl/csv/indexed/TemporalIndexedCSVDataSource.java

```diff
@@ -0,0 +1,175 @@
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.temporal.io.api.TemporalDataSource;
+import org.gradoop.temporal.io.impl.csv.functions.CSVLineToTemporalEdge;
+import org.gradoop.temporal.io.impl.csv.functions.CSVLineToTemporalGraphHead;
+import org.gradoop.temporal.io.impl.csv.functions.CSVLineToTemporalVertex;
+import org.gradoop.temporal.model.impl.TemporalGraph;
+import org.gradoop.temporal.model.impl.TemporalGraphCollection;
+import org.gradoop.temporal.model.impl.TemporalGraphCollectionFactory;
+import org.gradoop.temporal.model.impl.TemporalGraphFactory;
+import org.gradoop.temporal.model.impl.pojo.TemporalEdge;
+import org.gradoop.temporal.model.impl.pojo.TemporalGraphHead;
+import org.gradoop.temporal.model.impl.pojo.TemporalVertex;
+import org.gradoop.temporal.util.TemporalGradoopConfig;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A temporal graph data source for CSV files indexed by label.
+ * <p>
+ * The datasource expects files separated by label, e.g. in the following directory structure:
+ * <p>
+ * csvRoot
+ * |- vertices
+ *   |- person
+ *     |- data.csv    # contains all vertices with label 'Person'
+ *   |- university
+ *     |- data.csv    # contains all vertices with label 'University'
+ * |- edges
+ *   |- knows
+ *     |- data.csv    # contains all edges with label 'knows'
+ *   |- studyAt
+ *     |- data.csv    # contains all edges with label 'studyAt'
+ * |- graphs
+ *   |- myGraph
+ *     |- data.csv
+ * |- metadata.csv   # Meta data for all data contained in the graph
+ */
+public class TemporalIndexedCSVDataSource extends IndexedCSVDataSource implements TemporalDataSource {
+
+  /**
+   * Creates a new data source. The constructor creates a default HDFS configuration.
+   *
+   * @param csvPath root path of csv files
+   * @param config  gradoop configuration
+   */
+  public TemporalIndexedCSVDataSource(String csvPath, TemporalGradoopConfig config) {
+    super(csvPath, config, new Configuration());
+  }
+
+  /**
+   * Reads a {@link TemporalGraph} and converts it to a {@link LogicalGraph} via function
+   * {@link TemporalGraph##getLogicalGraph()}.
+   *
+   * @return a logical graph instance
+   * @throws IOException in case of an error while reading the graph
+   */
+  @Override
+  public LogicalGraph getLogicalGraph() throws IOException {
+    return getTemporalGraph().toLogicalGraph();
+  }
+
+  /**
+   * Reads a {@link TemporalGraphCollection} and converts it to a {@link GraphCollection} via function
+   * {@link TemporalGraphCollection##getGraphCollection()}.
+   *
+   * @return a graph collection instance
+   * @throws IOException in case of an error while reading the graph collection
+   */
+  @Override
+  public GraphCollection getGraphCollection() throws IOException {
+    return getTemporalGraphCollection().toGraphCollection();
+  }
+
+  @Override
+  public TemporalGraph getTemporalGraph() throws IOException {
+
```

Comment:
I don't see a problem with reusing `getTemporalGraphCollection()` here (like in `IndexedCSVDataSource`). Or create a new private method `readIndexedCSV()` that returns a `TemporalIndexedLayout` and use that in `getTemporalGraph()` and `getTemporalGraphCollection`.